### PR TITLE
Facet processor context queries method.

### PR DESCRIPTION
**Pull request by claygoddard**
Extract out context queries method for shared usage among facets.

### CODE REVIEW

**Comment by claygoddard** [comment]
Opening fs-solr PR momentarily to show usage. Improves on doc set intersection for computations.

File: solr/core/src/java/org/apache/solr/search/facet/FacetProcessor.java [hunk 1]
```diff
@@ -207,23 +208,22 @@ private void handleFilterExclusions() throws IOException {
...
+    qlist.add(rb.getQuery());
 
     // add the filters
     if (rb.getFilters() != null) {
-      for (Query q : rb.getFilters()) {
-        if (!excludeSet.containsKey(q)) {
-          qlist.add(q);
-        }
-      }
+      qlist.addAll(rb.getFilters());
```
Comment (bradbesserman) [inline solr/core/src/java/org/apache/solr/search/facet/FacetProcessor.java:226]: What's the reason for removing these excludeSet checks?

Comment (claygoddard) [reply solr/core/src/java/org/apache/solr/search/facet/FacetProcessor.java:226]: They aren't removed. They are handled in a single pass: https://github.com/fullstorydev/lucene-solr/pull/127/files#diff-bf535b312bde0f43bcc717fd98de9d6bfb4b63c2706d94f43bda03995a199debR212. I can separate that line out of you want?

Comment (bradbesserman) [reply solr/core/src/java/org/apache/solr/search/facet/FacetProcessor.java:226]: Oh, 🤦 .  No need, LG!

**Review by bradbesserman — APPROVED** [review]
LGTM!

**Review by hiteshk25 — APPROVED** [review]
LGTM

**Pull request merged**